config/initializers/content_security_policy.rb

```diff
@@ -1,30 +1,27 @@
 # Be sure to restart your server when you modify this file.
 
-# Define an application-wide content security policy
-# For further information see the following documentation
-# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+# Define an application-wide content security policy.
+# See the Securing Rails Applications Guide for more information:
+# https://guides.rubyonrails.org/security.html#content-security-policy-header
 
-Rails.application.config.content_security_policy do |policy|
-  policy.default_src :self
-  policy.font_src    :self, :data
-  policy.img_src     :self, :data
-  policy.object_src  :none
-  policy.script_src  :self
-  policy.style_src   :self
-  # If you are using webpack-dev-server then specify webpack-dev-server host
-  # policy.connect_src :self, :https, "http://localhost:3035", "ws://localhost:3035" if Rails.env.development?
+Rails.application.configure do
+  config.content_security_policy do |policy|
+    policy.default_src :self
+    policy.font_src    :self, :data
+    policy.img_src     :self, :data
+    policy.object_src  :none
+    policy.script_src  :self
+    policy.style_src   :self
+    # Specify URI for violation reports
+    # policy.report_uri "/csp-violation-report-endpoint"
+  end
 
-  # Specify URI for violation reports
-  # policy.report_uri "/csp-violation-report-endpoint"
-end
-
-# If you are using UJS then enable automatic nonce generation
-Rails.application.config.content_security_policy_nonce_generator = -> request { SecureRandom.base64(16) }
+  # Generate session nonces for permitted importmap and inline scripts
+  config.content_security_policy_nonce_generator = lambda do |request|
+    request.session[:csp_nonce] ||= SecureRandom.base64(16)
+  end
```

Comment:
This is the solution we settled on in umts/round_three#6599, not what was generated by the update task.